src/function.py

```diff
@@ -169,6 +189,20 @@ def _filter_log_lines(log_entry):
     return ret
 
 
+def _calculate_session_timeout():
```

Comment:
Shouldn't this take the function's configured timeout into consideration?

Reply:
It does, take a look at line `199`

Reply:
Doesn't appear to be checking `context.get_remaining_time_in_millis()`. Lambda tells us how much time we have left in the invocation, so a best effort based on that is probably ideal?

Reply:
Sorry for the confusion and thanks for raising this point. As discussed offline, though, we'll keep the current simpler approach and consider this one if we face issues in the future.